File: CPP/atbash.cpp

#include <iostream>
#include <string>

using namespace std;

string atbash(const string& text) {
    string result;

    for (int ch : text) {
        if (ch >= 'A' && ch <= 'Z') {
            result += 'Z' - (ch - 'A');
        }
        else if (ch >= 'a' && ch <= 'z') {
            result += 'z' - (ch - 'a');
        }
        else if (ch >= '0' && ch <= '9') {
            result += ch;
        }
        else {
            result += ch;
        }
    }

    return result;
}

int main() {

    //setlocale(LC_ALL, "Rus");

     /*string input_text = "Привет, world! 123456789";
     string output_text = atbash(input_text);

     cout << "Результат: " << output_text << endl;*/

    string input_text;
    cout << "Text:";

    getline(cin, input_text);

    string output_text = atbash(input_text);
    cout << "Encrypted text: " << output_text << endl;

    string decrypted_text = atbash(output_text);
    cout << "Decrypted text: " << decrypted_text << endl;
    return 0;
}

File: PYTHON/atbash.py
def atbash(text):
    result = ""
    for ch in text:
        if 'A' <= ch <= 'Z':
            result += chr(ord('Z') - (ord(ch)-ord('A')))
        elif 'a' <= ch <= 'z':
            result += chr(ord('z') - (ord(ch)-ord('a')))
        elif '0' <= ch <= '9':
            result += ch
        else:
            result+=ch

    return result

text = "Hello, мир! 1234567890"
output_text = atbash(text)
print("Зашифрованный текст: ", output_text)

decrypted_text = atbash(output_text) #дешифрование - обратное преобразование
print("Расшифрованный текст: ", decrypted_text)